From 34657ad5b1eeaecae047169f5e3617ba1ad52766 Mon Sep 17 00:00:00 2001
From: Peter Swain <pswain@marvell.com>
Date: Thu, 24 Jan 2019 01:38:10 -0800
Subject: [PATCH 063/138] mmc: cavium: CMD19/21 type correction

commit cbeecc0d045020372935d4982b5aabfa4cf8dd4b from
git@git.assembla.com:cavium/WindRiver.linux.git

The Octeon host controller assumes the return-type and
bus-activity type of commands, so it can boot from eMMC
without software assistance, and support minimal operation
with minimal firmware.
As some details differ between eMMC and SD-card dialects,
and new commands are added as standards evolve, exceptions
to this assumed ctype/rtype table may be forced as commands
are issued, via MIO_EMM_CMD[CTYPE_XOR,RTYPE_XOR] fields.

The driver's knowledge of the hardware's assumption about
two tuning commands is incorrect:
- BUSTEST_W (CMD19) is an output command;
- MMC_SEND_TUNING_BLOCK_HS200 (CMD21) is an R1 read command.

Now these can be issued without error.

Change-Id: I60086d3164aee116cb6b1daf496691430da35544
Signed-off-by: Peter Swain <pswain@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Reviewed-on: https://sj1git1.cavium.com/8545
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Reviewed-by: Chandrakala Chavva <cchavva@marvell.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/mmc/host/cavium.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/mmc/host/cavium.c b/drivers/mmc/host/cavium.c
index 7ebcd3b72af2..17532265ad8a 100644
--- a/drivers/mmc/host/cavium.c
+++ b/drivers/mmc/host/cavium.c
@@ -71,9 +71,9 @@ static struct cvm_mmc_cr_type cvm_mmc_cr_types[] = {
 	{0, 1},		/* CMD16 */
 	{1, 1},		/* CMD17 */
 	{1, 1},		/* CMD18 */
-	{3, 1},		/* CMD19 */
+	{2, 1},		/* CMD19 */
 	{2, 1},		/* CMD20 */
-	{0, 0},		/* CMD21 */
+	{1, 1},		/* CMD21 */
 	{0, 0},		/* CMD22 */
 	{0, 1},		/* CMD23 */
 	{2, 1},		/* CMD24 */
-- 
2.17.1

